core/app/models/spree/shipping_manifest.rb

```diff
@@ -7,7 +7,7 @@ def initialize(inventory_units:)
 
   def for_order(order)
     Spree::ShippingManifest.new(
-      inventory_units: @inventory_units.select { |iu| iu.order_id == order.id }
+      inventory_units: @inventory_units.select { |iu| iu.shipment.order_id == order.id }
```

Comment:
I'm slightly concerned that this change may introduce `N+1` queries (and tvdeyen's suggestion `2N+1`)